File: frontend/lib/types/ai.ts
/**
 * AI feature types — M16 infrastructure, consumed by M17 (Chat) and M19 (Story Intelligence).
 */

export interface Character {
  id: string;
  novel_id: string;
  name: string;
  description: string | null;
  traits: string[];
  first_chapter: number | null;
}

export interface EmbeddingChunk {
  id: string;
  chapter_id: string;
  chunk_index: number;
  content_preview: string;
  vector_id: string;
}

// M19: stored as JSONB on novels.relationship_graph
export interface RelationshipNode {
  id: string;
  name: string;
}

export interface RelationshipEdge {
  source: string;
  target: string;
  weight: number;
}

export interface RelationshipGraph {
  nodes: RelationshipNode[];
  edges: RelationshipEdge[];
}

// M19: stored as JSONB on novels.arc_timeline
export interface TimelineEvent {
  chapter_number: number;
  event_summary: string;
}

File: backend/app/models/ai.py
"""Pydantic schemas for AI/vector infrastructure (M16+)."""
from typing import Optional

from pydantic import BaseModel


class CharacterPublic(BaseModel):
    id: str
    novel_id: str
    name: str
    description: Optional[str] = None
    traits: list[str] = []
    first_chapter: Optional[int] = None
    model_config = {"from_attributes": True}


class EmbeddingChunkPublic(BaseModel):
    id: str
    chapter_id: str
    chunk_index: int
    content_preview: str
    vector_id: str
    model_config = {"from_attributes": True}
